src/Elrond/Codec.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Codec.h"
+
+#include "HexCoding.h"
+
+using namespace TW;
+using namespace TW::Elrond;
+
+void normalizeHexStringTopLevel(std::string& value);
+
+std::string Codec::encodeStringTopLevel(const std::string& value) {
+    std::string encoded = hex(TW::data(value));
+    return encoded;
+}
+
+std::string Codec::encodeUint64TopLevel(uint64_t value) {
+    std::string encoded = hex(value);
+    normalizeHexStringTopLevel(encoded);
+    return encoded;
+}
+
+std::string Codec::encodeBigIntTopLevel(const std::string& value) {
+    return encodeBigIntTopLevel(uint256_t(value));
+}
+
+// For reference, see https://docs.elrond.com/developers/developer-reference/elrond-serialization-format/#arbitrary-width-big-numbers.
+std::string Codec::encodeBigIntTopLevel(uint256_t value) {
+    // First, load the 256-bit value into a "Data" object.
+    auto valueAsData = Data();
+    encode256BE(valueAsData, value, 256);
+    
+    std::string encoded = hex(valueAsData);
+    normalizeHexStringTopLevel(encoded);
+    return encoded;
+}
+
+/// Normalizes a hex string. The passed string is mutated.
+/// E.g.: "A" -> "0A"; "00A" -> "0A".
+void normalizeHexStringTopLevel(std::string& value) {
```

Comment:
Could you please move this code to `HexCoding.h`? so others can reuse it

Reply:
Fixed, moved it (and added a short test).